### PR TITLE
chore: deprecate Custom Searcher [MD-504]

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description

Deprecate all custom searchers and methods. 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by MikhailKardash — APPROVED** [review]
LGTM

File: docs/model-dev-guide/api-guides/apis-howto/deepspeed/autotuning.rst [hunk 1]
```diff
@@ -4,6 +4,12 @@
  DeepSpeed Autotune: User Guide
 ################################
 
+.. important::
+
+   **Deprecation Notice**: DeepSpeed Autotune has been deprecated. This feature will be removed in a
+   future release. Please see :ref:`Search Methods <_search-methods>` for details on supported
+   preset searchers.
+
```
Comment (tara-det-ai) [inline docs/model-dev-guide/api-guides/apis-howto/deepspeed/autotuning.rst:12]: .. important::

   **Deprecation Notice**: DeepSpeed Autotune is deprecated and will be removed in an upcoming release. 
   For information on supported search methods, please visit to :ref:`search-methods`.

Comment (tara-det-ai) [reply docs/model-dev-guide/api-guides/apis-howto/deepspeed/autotuning.rst:12]: @azhou-determined i don't think the ref link renders with that extra "_" in front?

Comment (tara-det-ai) [reply docs/model-dev-guide/api-guides/apis-howto/deepspeed/autotuning.rst:12]: does it build locally?

Comment (tara-det-ai) [reply docs/model-dev-guide/api-guides/apis-howto/deepspeed/autotuning.rst:12]: i think all instances need to be changed

**Review by tara-det-ai — COMMENTED** [review]
suggested edits

**Pull request merged**